File: 832_flip_image.java
class Solution {
    public int[][] flipAndInvertImage(int[][] A) {
        int len = A[0].length;
        for(int[] row : A){
            for(int i = 0; i<(len + 1)/2; i++){
                int temp = row[i];
                row[i] = ~row[len - i - 1] + 2;
                row[len - i - 1] = ~temp + 2;
            }
        }
        return A;        
    }
}

File: 832_flip_image.py
# -*- coding: utf-8 -*-
"""
Created on Sat Sep 21 07:27:44 2019

@author: kenzo
"""

class Solution:
    def flipAndInvertImage(self, A: List[List[int]]) -> List[List[int]]:
        for i in range(len(A)):
            A[i] = [~x + 2 for x in A[i][::-1]]
        return A